packages/dd-trace/src/dogstatsd.js

```diff
@@ -153,7 +153,67 @@ class NoopDogStatsDClient {
   flush () { }
 }
 
+// This is a simplified user-facing proxy to the underlying DogStatsDClient instance
+class CustomMetrics {
+  constructor (options) {
+    this.dogstatsd = new DogStatsDClient(options)
+  }
+
+  increment (stat, value = 1, tags) {
+    return this.dogstatsd.increment(
+      stat,
+      value,
+      CustomMetrics.tagTranslator(tags)
+    )
+  }
+
+  decrement (stat, value = 1, tags) {
+    return this.dogstatsd.increment(
+      stat,
+      value * -1,
+      CustomMetrics.tagTranslator(tags)
+    )
+  }
+
+  gauge (stat, value, tags) {
+    return this.dogstatsd.gauge(
+      stat,
+      value,
+      CustomMetrics.tagTranslator(tags)
+    )
+  }
+
+  distribution (stat, value, tags) {
+    return this.dogstatsd.distribution(
+      stat,
+      value,
+      CustomMetrics.tagTranslator(tags)
+    )
+  }
+
+  flush () {
+    return this.dogstatsd.flush()
+  }
+
+  /**
+   * Exposing { tagName: 'tagValue' } to the end user
+   * These are translated into [ 'tagName:tagValue' ] for internal use
+   */
```

Comment:
Any reason not to support array form externally too and just do the conversion if it's not _already_ an array?

Reply:
It's convenient for us but I can't imagine an end user ever wanting to provide data in that format. All other products also use key/value object pairs. We can always support arrays in the future if anyone requests it.